std/typecons.d

```diff
@@ -2064,6 +2075,46 @@ if (is(T == class) || is(T == interface) || isDynamicArray!T || isAssociativeArr
     static assert(!__traits(compiles, &r.get()));
 }
 
+@system unittest // issue 18615; object.opEquals is not @safe
+{
+    class CustomOpEq
+    {
+        int x;
+        override bool opEquals(Object rhsObj)
+        {
+            const rhs = cast(const(CustomOpEq)) rhsObj;
+            return this.x == rhs.x;
```

Comment:
FYI: in general this should be:

```d
if (auto rhs = cast(const(CustomOpEq)) rhsObj)
    return this.x == rhs.x;
else
    return false;
```

(though obviously that's not necessary for this test)

Reply:
Thanks, good call, I'll use your code. All unittests should be written like examples even when this particular one isn't exported as documentation.